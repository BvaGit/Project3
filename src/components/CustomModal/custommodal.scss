@import '../../utils/variables.scss';

.modal{
    position: fixed;
    z-index: 1;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    background-color: rgba($title-color, 0.8);
}
.modal-content{
    background-color: $background;
    margin: 15% auto;
    padding: 20px;
    width: 30%;
    align-items: center;
}
.close{
    color: $button-color;
    float: right;
    font-size: 20px;
    font-weight: bold;
    cursor: pointer;
}
.modal-header{
    text-align: center;
    font-size: 20px;
    margin-bottom: $margin_md;
}