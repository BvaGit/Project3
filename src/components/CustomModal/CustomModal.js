import React from "react";
import { FormattedMessage } from "react-intl";

import Portal from "../Portal";

import "./custommodal.scss";

function CustomModal({
  languageValue: langValue,
  handlecloseModal,
  header,
  content,
}) {
  return (
    <Portal>
      <div className="modal">
        <div className="modal-content">
          <span className="close" onClick={handlecloseModal}>
            &times;
          </span>
          <div className="modal-header">
            <FormattedMessage
              id={header}
              defaultMessage="First name"
              value={{ langValue }}
            >
              {header}
            </FormattedMessage>
          </div>
          {content}
        </div>
      </div>
    </Portal>
  );
}

export default CustomModal;
