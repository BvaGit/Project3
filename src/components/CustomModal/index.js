import { connect } from 'react-redux';
import CustomModal from "./CustomModal";
import { languageValue } from '../../store/localization/selectors';

const mapStateToProps = (state) => ({
    languageValue: languageValue(state)
})

export default connect(mapStateToProps, null)(CustomModal);