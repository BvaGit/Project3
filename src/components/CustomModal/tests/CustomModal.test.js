import React from 'react';
import { shallow, mount } from 'enzyme';
import CustomModal from '../CustomModal';


describe('CustomModal', () => {
    it("Should match snapshot", () => {
        const component = shallow(<CustomModal />)
        expect(component).toMatchSnapshot()
    })
})