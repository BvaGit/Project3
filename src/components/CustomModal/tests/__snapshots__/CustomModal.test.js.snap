// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal Should match snapshot 1`] = `ShallowWrapper {}`;
